packages/common-library/src/common_library/async_tools.py

```diff
@@ -62,3 +65,24 @@ async def maybe_await(
         return await obj
     assert not isawaitable(obj)  # nosec
     return obj
+
+
+async def cancel_and_wait(task: asyncio.Task) -> None:
+    """Cancels the given task and waits for it to finish.
+
+    Accounts for the case where the tasks's owner function is being cancelled
+    """
+    task.cancel()
+    try:
+        # NOTE shield ensures that cancellation of the caller function won’t stop you
+        # from observing the cancellation/finalization of task.
+        await asyncio.shield(task)
+    except asyncio.CancelledError:
+        if not task.cancelled():
```

Comment:
nice!, but please replace also `cancel_wait_task` then...

Reply:
Ahh you mean there is one in `servicelib.async_utils.cancel_wait_task`!